packages/twenty-server/src/engine/api/graphql/workspace-query-runner/listeners/entity-events-to-db.listener.ts

```diff
@@ -50,16 +50,18 @@ export class EntityEventsToDbListener {
   }
 
   private async handle(payload: WorkspaceEventBatch<ObjectRecordBaseEvent>) {
-    payload.events = payload.events.filter(
+    const newPayload = { ...payload };
```

Comment:
let's avoid creating referenced objects. You have fixed this particular case but we are STILL modifying references:
<img width="333" alt="image" src="https://github.com/user-attachments/assets/433c12c7-4edd-441f-9b3e-34c8a0f5b8ed">

this pattern could lead to side effects issues like the one you faced


Instead: 
const filteredEvents = payloads.events.filter(...) 

this.messageQueueService.add({ ...payload, events: filteredEvents })